src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp

```diff
@@ -970,25 +970,30 @@ void ShenandoahBarrierC2Support::call_lrb_stub(Node*& ctrl, Node*& val, Node* lo
 
   address calladdr = NULL;
   const char* name = NULL;
-  switch (kind) {
-    case ShenandoahBarrierSet::AccessKind::NATIVE:
+  bool is_strong  = ShenandoahBarrierSet::is_strong_access(decorators);
+  bool is_weak    = ShenandoahBarrierSet::is_weak_access(decorators);
+  bool is_phantom = ShenandoahBarrierSet::is_phantom_access(decorators);
+  bool is_native  = ShenandoahBarrierSet::is_native_access(decorators);
+  bool is_narrow  = LP64_ONLY(UseCompressedOops &&) !is_native;
+  if (is_strong) {
+    if (is_narrow) {
+      calladdr = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_strong_narrow);
+    } else {
+      calladdr = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_strong);
+    }
+    name = "load_reference_barrier_strong";
```

Comment:
Why not separate "load_reference_barrier_strong" and  "load_reference_barrier_strong_narrow"?

Reply:
Dunno. Could do that. Within one instance of JVM, we would never see both narrow/non-narrow versions, though.

Reply:
I would be more straight-forward looking, AFAICS. Is there any code that actually matches those strings?

Reply:
I don't think any code matches those strings. I will change them to be more specific.